### PR TITLE
ACMS-1289: Refactor role permissions.

### PR DESCRIPTION
**Pull request by vishalkhode1**
**Motivation**
Refactor role permissions,

**Proposed changes**
Refactor Roles Permissions, move permission to specific modules.
Acquia CMS Common
- Removed permissions related to site studio, toolbar and tour from roles config file.
Acquia CMS Site Studio
- Added roles permissions related to site studio in hook_install and hook_update
Acquia CMS Toolbar
- Added roles permissions related to toolbar in hook_install and hook_update
Acquia CMS Tour
- Added roles permissions related to tour in hook_install and hook_update

**Alternatives considered**
PHP Unit test related to role permissions i.e. [BasicPermissionsTest](https://github.com/acquia/acquia_cms_common/blob/1.4.6/tests/src/Functional/BasicPermissionsTest.php) will fail.

**Testing steps**
New Site:
- Install site with this branch
- Roles should have the permissions listed in [BasicPermissionsTest](https://github.com/acquia/acquia_cms_common/blob/1.4.6/tests/src/Functional/BasicPermissionsTest.php)

Existing Site:
- Update codebase to this branch
- Run database update 
- - Roles should have the permissions listed in [BasicPermissionsTest](https://github.com/acquia/acquia_cms_common/blob/1.4.6/tests/src/Functional/BasicPermissionsTest.php)

- Other PHPUnit tests should pass.
- Manual Testing.

**Merge requirements**
- [ ] _Major change_,
- [ ] Manual testing by a reviewer

### CODE REVIEW

**Review by rajeshreeputra — COMMENTED** [review]
From modules/acquia_cms_tour 1.patch and 2.patch file need to be removed, apart from that everything else looks good.

File: modules/acquia_cms_site_studio/acquia_cms_site_studio.install [hunk 1]
```diff
@@ -183,15 +331,142 @@ function acquia_cms_site_studio_update_8003() {
...
  */
-function acquia_cms_site_studio_uninstall() {
-  $config = \Drupal::service('config.factory')->getEditable('field.field.node.page.body');
-  if (!empty($config)) {
-    $config->set('label', 'Body');
-    $config->set('description', '');
-    $config->save();
+function acquia_cms_site_studio_update_8004() {
+  $moduleHandler = \Drupal::service('module_handler');
+  if ($moduleHandler->moduleExists('acquia_cms_site_studio')) {
```
Comment (vishalkhode1) [inline modules/acquia_cms_site_studio/acquia_cms_site_studio.install:338]: @rajeshreeputra Same as mentioned below i.e de-dupe code and condition check not required.
Also, can we make the code more readable ? Ex: 

```
// Define basic permission.
$site_studio_basic_permissions = [
  'access markup',
  'access seo',
  'access styles',
  'access tokens',
  //....and so on.
];

$basic_component_perimissions = [
  'access accordion_sections cohesion_helper_category group',
  'access card_sections cohesion_helper_category group',
  'access cpt_cat_accordion_components cohesion_component_category group',
  'access cpt_cat_basic_components cohesion_component_category group',
];

$role_permissions = [
  'content_author' => $site_studio_basic_permissions,
  'developer' => array_merge($site_studio_basic_permissions, $basic_component_perimissions),
  //.. and so on for other roles.
];
```

File: modules/acquia_cms_tour/acquia_cms_tour.install [hunk 2]
```diff
@@ -5,9 +5,62 @@
...
 function acquia_cms_tour_update_8001() {
   \Drupal::state()->set('existing_site_acquia_cms', TRUE);
 }
+
+/**
+ * Update role permissions.
+ */
+function acquia_cms_tour_update_8002() {
+  $moduleHandler = \Drupal::service('module_handler');
+  if ($moduleHandler->moduleExists('acquia_cms_tour')) {
```
[the remark this answers is not in the document]
Comment (rajeshreeputra) [reply modules/acquia_cms_tour/acquia_cms_tour.install:46]: done.

File: modules/acquia_cms_tour/2.patch [hunk 3]
```diff
@@ -0,0 +1,83 @@
+diff --git a/acquia_cms_tour.install b/acquia_cms_tour.install
```
[the remark this answers is not in the document]
Comment (rajeshreeputra) [reply modules/acquia_cms_tour/2.patch:1]: done.

File: modules/acquia_cms_tour/1.patch [hunk 4]
```diff
@@ -0,0 +1,67 @@
+diff --git a/acquia_cms_tour.install b/acquia_cms_tour.install
```
[the remark this answers is not in the document]
Comment (rajeshreeputra) [reply modules/acquia_cms_tour/1.patch:1]: done.

File: modules/acquia_cms_toolbar/acquia_cms_toolbar.install [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    }
+  }
+}
+
+/**
+ * Update role permissions.
+ */
+function acquia_cms_toolbar_update_8001() {
+  $moduleHandler = \Drupal::service('module_handler');
+  if ($moduleHandler->moduleExists('acquia_cms_toolbar')) {
```
[the remark this answers is not in the document]
Comment (rajeshreeputra) [reply modules/acquia_cms_toolbar/acquia_cms_toolbar.install:35]: done.

File: modules/acquia_cms_site_studio/acquia_cms_site_studio.install (hunk 1 above)
Comment (rajeshreeputra) [reply modules/acquia_cms_site_studio/acquia_cms_site_studio.install:338]: added comment above permissions

File: modules/acquia_cms_tour/acquia_cms_tour.install [hunk 6]
```diff
@@ -5,9 +5,62 @@
...
+    $roles = Role::loadMultiple();
```
[the remark this answers is not in the document]
Comment (rajeshreeputra) [reply modules/acquia_cms_tour/acquia_cms_tour.install]: done.

**Pull request merged**